src/Blockcore.Indexer.Cirrus/Storage/Mongo/SmartContracts/Dao/VoteLogReader.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Blockcore.Indexer.Cirrus.Client.Types;
+using Blockcore.Indexer.Cirrus.Storage.Mongo.Types;
+
+namespace Blockcore.Indexer.Cirrus.Storage.Mongo.SmartContracts.Dao;
+
+class VoteLogReader : ILogReader
+{
+   public bool CanReadLogForMethodType(string methodType) => methodType == "Vote";
+
+   public bool IsTheTransactionLogComplete(LogResponse[] logs) => true;
+
+   public void UpdateContractFromTransactionLog(CirrusContractTable contractTransaction,
+      DaoContractComputedTable computedTable)
+   {
+      if (contractTransaction.Logs.Length == 0)
+      {
+         //TODO need to handle this wierd issue (example transaction id - 5faa9c5347ba378ea1b4dd9e957e398867f724a6ba4951ed65cde6529dbfd6a0)
+         return;
+      }
+
+      int id = (int)(long)contractTransaction.Logs.First().Log.Data["proposalId"];
+      bool voteYesNo = (bool)contractTransaction.Logs.First().Log.Data["vote"];
+      string voterAddress = (string)contractTransaction.Logs.First().Log.Data["voter"];
+
+      var proposal = computedTable.Proposals.SingleOrDefault(_ => _.Id == id);
+
+      if (proposal is null)
+      {
+         throw new InvalidOperationException(
```

Comment:
What does this mean? we block the indexer? 
I suppose if this should not happen we should throw and not let the indexer continue till we fix this bug